frontend/src/plugin/lib.ts

```diff
@@ -0,0 +1,141 @@
+/**
+ * The lib.ts file should carry the bits to be used by plugins whereas
+ * the index.ts should have the functions that Headlamp itself needs for
+ * loading the plugins.
+ */
+
+/**
+ * ## Simple plugin example
+ *
+ *
+ * @example
+ * // Here's a very simple JavaScript plugin example, that does nothing.
+ * import { Headlamp } from '@kinvolk/headlamp-plugin/lib';
```

Comment:
How about just `import { registerPlugin } from '@kinvolk/headlamp-plugin/lib';` instead? I don't think the Headlamp class is needed here.

Reply:
I wonder if it's useful to have it as a namespace kind of helper.
We could also default export a list of plugins, like:

```
export default PLUGINS = [
   MyPlugin
];
```